### ISSUE TITLE
Sharing a whole hard disk drive content doesn't work

### ISSUE THREAD

**Issue by ewauq** [Nicotine-Plus/nicotine-plus#999]
Nicotine+ version:  3.0.0
Operating System/Distribution: Windows 10 Professionnal 20H2

### Describe the bug
Sharing an HDD from its root (like M:\) doesn't share any file. I have to specify a folder (M:\Folk for example) to get the file sharing working properly. 

So I have to add all folders at the HDD root one by one to share files.

The log console shows:
```
23:06:42 Rescanning normal shares...
23:06:43 0 folders found before rescan, rebuilding...
23:06:44 0 folders found after rescan
```

### Expected behavior
Adding a HDD root folder shares all its folders and subfolders content.

### Steps to reproduce the bug

- Preferences > Transfers > Shares > Public Shares
- Click Add... and select a HDD without selecting any of its folders
- Give a Virtual Name (like Collection) and click OK

File scanning doesn't find any folder.

I wanted to show the process with a gif file but there is too many personnal data on the capture.

**Comment by mathiascode** [comment]
It seems like Windows root directories have the FILE_ATTRIBUTE_HIDDEN attribute, which Nicotine+ respects. Perhaps we need to add a workaround for this special case.

**Comment by ewauq** [comment]
Thank you for the answer. The current workaround is to share folders directly, so I can use Nicotine+, this is not a problem, I don't have a lot of folders. Glad to see you found the cause :)

**Comment by mathiascode** [comment]
There was a bug where the root directory would be scanned, but not shared properly. This will be fixed in 3.0.2.
